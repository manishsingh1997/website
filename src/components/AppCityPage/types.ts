export type City = {
  slug: string,
  city: string,
  county: string,
  state:
    | 'AL' | 'AK' | 'AZ' | 'AR' | 'CA' | 'CO' | 'CT' | 'DE' | 'DC' | 'FL' | 'GA' | 'HI' | 'ID'
    | 'IL' | 'IN' | 'IA' | 'KS' | 'KY' | 'LA' | 'ME' | 'MD' | 'MA' | 'MI' | 'MN' | 'MS' | 'MO'
    | 'MT' | 'NE' | 'NV' | 'NH' | 'NJ' | 'NM' | 'NY' | 'NC' | 'ND' | 'OH' | 'OK' | 'OR' | 'PA'
    | 'RI' | 'SC' | 'SD' | 'TN' | 'TX' | 'UT' | 'VT' | 'VA' | 'WA' | 'WV' | 'WI' | 'WY',
  address?: string,
  phone: string,
  header: {
    title: string,
    bullets: string[],
  },
  blog: {
    articles: Array<{
      url: string,
      img: string,
      description: string,
    }>,
  },
  license: {
    url?: string,
  },
  licenses?: string[],
  packages: {
    title: string,
    description: string,
    data: Record<'wood' | 'vinyl' | 'chain-link' | 'box-wire', Array<{
      title: string,
      description: string,
      img: string,
    }>>,
  },
  projects: Array<{
    url?: string,
    img: string,
    label: string,
  }>,
  product: {
    type: 'Fence',
  },
  rating: Record<'Angi' | 'BBB' | 'Google' | 'Thumbtack' | 'Yelp', string>,
  review: Record<'Yelp' | 'Google', {
    title: string,
    name?: string,
    description?: string,
    address?: string,
    img?: string,
    url?: string,
  }>,
  regulations: {
    title: string,
    description: string,
    pdf?: string,
    img?: string,
    faq: Record<'front-yard' | 'side-yard' | 'rear-max-height' | 'setback' | 'corner-lot-rules', string>,
  },
};
