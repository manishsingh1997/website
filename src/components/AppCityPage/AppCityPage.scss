@import '~@ergeon/core-components/src/components/variables';

@import './MainBanner.scss';

.equal-height-JlocK {
  display: flex !important;
  align-items: stretch;
  justify-content: stretch;
}

.AppCityPage .CityReviews {
  $offset: $h-40;

  &-container {
    margin: $v-64 (-1 * $offset) 0;

    @include phone() {
      margin: $v-24 (-1 * $offset);
    }
  }

  &-review {
    position: relative;
    display: flex;
    flex: 1;
    align-items: flex-start;
    margin: 0 $offset $v-64;
  }

  &-imgContainer {
    $image-container-max-width: 154px;
    $image-container-h-offset: $h-5;

    @include border-soft;

    width: $image-container-max-width;
    margin-right: $h-20;
    padding: $v-6 $image-container-h-offset;
    background: $white;

    img {
      width: $image-container-max-width - $image-container-h-offset * 2;
    }

    @include phone {
      position: absolute;
      bottom: -1 * $v-30;
      right: 0;
      margin-right: 0;
    }
  }

  &-description {
    margin: $v-12 0;
    color: $black-3;
    font-style: italic;
  }

  .small-text + .small-text {
    margin-top: $v-6 / 2;
  }
}

.AppCityPage .RecentProjects {
  $offset: $h-40;

  margin: $v-64 auto;

  &-container {
    margin: $v-30 (-1 * $offset) $v-16;

    @include phone() {
      display: flex;
      flex-direction: row;
      justify-content: space-between;
      overflow-x: scroll;
      -ms-overflow-style: none;
      scrollbar-width: none;
      width: auto;
    }

    &::-webkit-scrollbar {
      display: none;
    }
  }

  &-project {
    margin: 0 $offset;

    @include phone() {
      margin: 0 $h-10;
    }
  }

  &-project:first-child {
    margin-left: $offset;
  }

  &-project:last-child {
    margin-right: $offset;
  }

  &-projectCaption {
    margin-top: $v-16;

    @include phone() {
      margin-top: $v-6;
    }
  }

  img {
    width: 100%;
    max-height: 360px;
    border-radius: $corner-soft;

    @include phone() {
      width: $h-120 * 2 + $h-60;
      max-height: 200px;
    }
  }

  a {
    color: inherit;
    font-weight: inherit;

    &:hover {
      color: darken($brand-color, 15%);
    }
  }
}

.AppCityPage .Packages {
  margin: $v-64 auto;

  &-container {
    grid-template-columns: calc(25% - 15px) calc(25% - 15px) calc(25% - 15px) calc(25% - 15px);
    margin-bottom: $v-48;

    @include phone {
      grid-template-columns: calc(50% - 15px) calc(50% - 15px);
    }
  }

  &-item {
    padding: $v-20 $h-20;

    img {
      width: 100%;
      margin-bottom: $v-12;
    }
  }

  &-title {
    margin-bottom: $v-6;
  }

  &-description {
    color: $black-3;
  }
}

.AppCityPage .CityBlog {
  $offset: $h-10;
  $column-width: 100% / 3;

  margin: $v-64 auto;

  &-container {
    grid-template-columns: calc(#{$column-width} - 15px) calc(#{$column-width} - 15px) calc(#{$column-width} - 15px);
    margin-top: $v-30;

    @include phone {
      display: block;
    }
  }

  &-article {
    display: flex;
    flex-direction: column;
    width: 100%;
    margin: 0;
    padding: $v-20 $h-20;

    p {
      flex: 1;
      font-weight: normal;
      color: $black-2;
    }

    @include phone {
      display: flex;
      flex-direction: row;
      border-radius: 0;
      border-left: 0;
      border-right: 0;
    }
  }

  &-img {
    $img-abs-width: 340;
    $img-abs-height: 239;

    width: 100%;
    margin-bottom: $v-20;
    padding-bottom: $img-abs-height / $img-abs-width * 100%;
    background-size: cover;

    @include phone {
      $pic-size: $v-64;

      width: $pic-size;
      height: $pic-size;
      margin-right: $h-10;
      padding-bottom: 0;
      border-radius: $corner-default;
    }
  }

  &-content {
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: space-between;
  }

  &-link {
    display: inline-block;
    padding-top: $v-6;

    @include phone {
      display: none;
    }
  }

  .equal-height-JlocK {
    @include phone {
      height: auto !important;
      margin: 0 (-1 * $h-20);

      &:not(:first-child) .CityBlog-article {
        border-top: 0;
      }
    }
  }
}

.mobile-length {
  display: none;

  @include phone() {
    display: block;
  }
}

.desktop-length {
  display: block;

  @include phone() {
    display: none;
  }
}
