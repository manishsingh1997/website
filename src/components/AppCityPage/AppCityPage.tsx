import React from 'react';

import NotFoundPage from '../NotFoundPage';
import {City} from './types';

export type AppCityPageProps = {
  city: City,
};

const AppCityPage = (props: AppCityPageProps) => {
  const {city} = props;

  console.log(city); // eslint-disable-line

  // NOTE: it will be replaced with actual content in upcoming PRs
  return <NotFoundPage />;
};

export default AppCityPage;
