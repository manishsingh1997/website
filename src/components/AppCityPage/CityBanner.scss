@import '~@ergeon/core-components/src/components/variables';

.CityBanner {
    $form-desktop-max-width: $h-60 * 6 +$h-20;
    $form-mobile-top: 520px;
    $logo-menu-bar-size: $v-64;
    $logo-menu-bar-size-mobile: $v-48;
    $bottom-margin-mobile: ($v-64 + 5) * 4;
    $wrapper-max-height-mobile: 719px;
    $content-max-width: 650px;

    @mixin form-background {
        border-radius: $corner-soft;
        background-color: $gray-6;
        border: 1px solid $gray-4;
        padding: 0;
    }

    @mixin font-appearance() {
        background-color: $white;
        border-radius: $corner-soft;
        padding: 0;
    }

    @mixin color-icon($color: $white) {
        svg path {
            fill: $color;
        }
    }

    @mixin icon-dimension($width, $height, $margin: 0) {
        svg {
            width: $width;
            height: $height;
            margin: $margin;
        }
    }

    @mixin break-text-mobile {
        br {
            display: none;

            @include phone {
                display: unset;
            }
        }
    }

    opacity: 1;
    background: linear-gradient(rgba(1, 64, 169, 1) 0%, rgba(42, 144, 229, 1) 52%, rgba(156, 177, 244, 1) 100%);
    color: $white;
    margin-top: $logo-menu-bar-size;

    @include phone {
        margin: $logo-menu-bar-size-mobile 0 $bottom-margin-mobile 0;
    }

    &-wrapper {
        display: flex;
        justify-content: space-between;
        padding: $v-124 - $v-16 $v-48;

        @include tablet {
            padding: $v-64 - 8 $h-20 + $h-10;
            flex-direction: column;
            justify-content: flex-start;
            align-items: center;
        }

        @include phone {
            max-height: $wrapper-max-height-mobile;
            height: 100%;
            padding: $v-48 $h-20 + $h-10;
        }
    }

    &-content {
        max-width: $content-max-width;
    }

    &-desc {
        margin-top: $v-42 + 2;

        @include tablet {
            text-align: center;
            margin-top: 0;
        }

        &-license {
            color: $white;
            font-weight: 500;
            margin-top: $v-42;
            cursor: pointer;
        }
    }

    &-info {
        .h2 {
            margin: $v-12 0;
            max-width: $h-40 * 11;
            width: 100%;
            font-weight: 700;
            @include tablet {
                margin: $v-12 0 $v-20;
                max-width: unset;
                line-height: $h4-line-height;
                font-weight: 700;
                font-size: $h4-size;
            }
        }

        .h4 {
            font-weight: 500;
            @include tablet {
                font-size: $font-regular-size;
            }
        }

        .h2,
        .h4 {
            @include break-text-mobile();
            color: inherit;
        }

        h3.h4 {
            @include tablet {
                font-size: $h6-size;
            }
        }

        h3:last-child {
            @include phone {
                margin-top: $v-12;
            }
        }
    }

    &-call {
        $icon-size: $v-16;
        $icon-margin: 0 $h-5 + 2 0;

        display: none;

        @include tablet {
            @include color-icon($white);
            @include icon-dimension($icon-size, $icon-size, $icon-margin);
            max-width: $h-40 * 7;
            width: 100%;
            display: block;
            margin: $v-30 + 4 auto 0;
        }

        @include phone {
            margin-top: $v-24;
        }

        button {
            width: 100%;
        }
    }


    .BannerRating {
        @include tablet {
            max-width: 184px;
            width: 100%;
            margin: $v-24 auto;
            font-size: $font-mobile-size;
            line-height: 1.42;
            text-align: center;
        }

        .stars-group {
            @include tablet {
                margin-top: $v-6;
            }
        }

        &-stars,
        &-ratingText {
            @include tablet {
                justify-content: center;
                align-items: center;

                .star,
                .star svg {
                    // important is used here to overwrite core-components important
                    width: $h-20 + 3 !important;
                    height: $v-24 - 2 !important;
                }
            }
        }
    }

    p.is-tablet {
        @include tablet {
            margin-bottom: $v-30;
        }

        @include phone {
            margin-bottom: $v-24;
        }
    }

    &-form {
        max-width: $form-desktop-max-width;
        width: 100%;

        @include tablet {
            max-width: $h-40 * 8;
        }

        @include phone {
            position: absolute;
            margin: auto;
            top: $form-mobile-top;
            left: $h-20 + $h-10;
            right: $h-20 + $h-10;

        }

        .TellUsForm {
            @include form-background();

            .react-select__indicator-separator {
                display: none;
            }

            .erg-input--text {
                background-color: $white;
            }

            &-Wrapper {
                padding: $v-24;
                width: 100%;
            }

            h3 {
                font-size: $h4-size;
                line-height: $h4-line-height;
                text-align: left;
            }

            &-Text {
                font-size: $font-mobile-size;
                line-height: $line-height-mobile;
                text-align: left;
                padding: 0;
            }

            &-Success {
                @include form-background();
                padding: $v-24;
                margin: 0;
                width: 100%;
            }
        }
    }

    .is {
        &-grey {
            opacity: 0.75;
        }

        &-action {
            background-color: $orange-action;
        }

        &-tablet {
            display: none;

            @include tablet {
                display: block;
            }
        }
    }
}
