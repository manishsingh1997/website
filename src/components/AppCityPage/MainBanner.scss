@import '~@ergeon/core-components/src/components/variables';

.AppCityPage .MainBanner {
  $font-color: $white;

  $icon-size: $v-24;
  $icon-offset: $h-10;
  $title-mobile-line-height: 1.36;
  $items-mobile-line-height: 1.43;
  $form-desktop-max-width: $h-60 * 6;

  @mixin base-font {
    font-size: $h4-size;
    font-family: $font-title;
    line-height: $h4-line-height;
    color: $font-color;
  }

  @mixin form-background {
    background-color: $white;
    border-radius: $corner-soft;
    padding: 0;
  }

  @mixin color-icon($color: $font-color) {
    svg path {
      fill: $color;
    }
  }

  @mixin icon() {
    @include color-icon();

    width: $icon-size;
    height: $icon-size;
    margin-right: $icon-offset;

    svg {
      width: 100%;
      height: 100%;
    }
  }

  @mixin underlay-block($size, $color) {
    position: absolute;
    display: block;
    top: 50%;
    right: 50%;
    margin: (-1 * $size / 2) (-1 * $size / 2) 0 0;
    z-index: -1;
    width: $size;
    height: $size;
    background: $color;
  }

  padding: $v-64 0;
  background-image: url('../../assets/fence-background.jpeg');
  background-size: cover;
  background-position: top;

  &.is-StateTX {
    background-image: url('../../assets/texas-fence-background.jpeg');
  }

  @include phone() {
    padding: $v-48 0;
  }

  &-content {
    flex: 2;
    max-width: $screen-sm;
    margin-right: $h-20;
  }

  &-form {
    max-width: $form-desktop-max-width;
  }

  &-title {
    color: $font-color;

    @include phone() {
      font-size: $h4-size;
      line-height: $title-mobile-line-height;
    }
  }

  &-list {
    padding: $v-24 0;
  }

  &-listItem {
    @include base-font();

    padding-left: $icon-size + $icon-offset;

    &::before {
      display: none;
    }

    .icon {
      @include icon();

      margin-left: -1 * ($icon-size + $icon-offset);
      padding: $corner-default;
    }

    @include phone() {
      font-size: $font-regular-size - 2;
      line-height: $items-mobile-line-height;
    }
  }

  &-callLink {
    @include base-font();

    display: flex;
    align-items: center;

    @include phone() {
      font-size: $font-regular-size;
      line-height: $items-mobile-line-height;
    }

    &:hover {
      text-decoration: underline;
    }

    .icon {
      @include icon();
      @include color-icon($green-1);

      position: relative;
      z-index: 1;

      &::before {
        @include underlay-block($icon-size / 2, $font-color);

        content: '';
      }
    }
  }

  .TellUsForm {
    @include form-background();

    &-Wrapper {
      padding: $v-24;
      width: 100%;
    }

    h3 {
      font-size: $h4-size;
      line-height: $h4-line-height;
      text-align: left;
    }

    &-Text {
      font-size: $font-mobile-size;
      line-height: $line-height-mobile;
      text-align: left;
      padding: 0;
    }
  }

  // this is for the Thanks confirmation area
  .request-quote-page {
    @include form-background();
    padding: $v-24;
  }
}
