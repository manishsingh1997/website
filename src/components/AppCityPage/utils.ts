import {City} from './types';

export const getAsset = (src: string, ext: 'jpeg' | 'pdf') => {
  const filename = src.match(/file\/d\/([^/]+)/)?.[1];
  if (!filename) return src;
  return `https://s3.us-west-2.amazonaws.com/${process.env.S3_BUCKET}/cities_photos/${filename}.${ext}`;
};

export const makePhoneLink = (phone: string, options = {countryCode: '+1'}) => {
  let link = phone.replace(/[^\d+]/g, '');
  if (!/^\+/.test(link)) {
    link = `${options.countryCode}${link}`;
  }
  return `tel:${link}`;
};

export const formatFooterLicenses = (licenses?: City['licenses'], licenseUrl?: string) => {
  if (!licenses) return null;
  return licenses.map((license) => {
    return {
      name: `License #: ${license}`,
      url: licenseUrl,
    };
  });
};
