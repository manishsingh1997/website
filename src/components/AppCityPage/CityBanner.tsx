import React from 'react';
import {ReactSVG} from 'react-svg';

import {Button} from '@ergeon/core-components';
import IconPhone from '@ergeon/core-components/src/assets/icon-call.svg';

import TellUsForm from '../../containers/TellUsForm';
import BannerRating from './components/BannerRating';

import './CityBanner.scss';

type CityBannerProps = {
  city: string;
};

const CityBanner = (props: CityBannerProps) => {
  const {city} = props;
  return (
    <section className="CityBanner" data-testid="city-banner">
      <section className="CityBanner-wrapper wrapper-1180">
        <section className="CityBanner-content">
          <aside className="CityBanner-desc">
            <aside className="CityBanner-info">
              <h1 className="h4 is-grey">#1 Fence Company in {city}</h1>
              <h2 className="h2">
                Finally a fence company <br />
                you can trust
              </h2>
              <h3 className="h4">
                Get your fence installed <br />
                by a world-class expert.
              </h3>
              <h3 className="h4">Zero hassle. Fair prices. No hidden fees.</h3>
            </aside>
            <a className="CityBanner-call" href="tel:+16503004854">
              <Button className="is-action" flavor="action" taste="solid">
                <ReactSVG src={IconPhone} /> Call us now
              </Button>
            </a>
            <BannerRating />
          </aside>
        </section>
        <p className="is-tablet">or</p>
        <section className="CityBanner-form">
          <TellUsForm />
        </section>
      </section>
    </section>
  );
};

export default CityBanner;
