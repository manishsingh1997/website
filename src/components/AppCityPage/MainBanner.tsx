import React from 'react';

import { ReactSVG } from 'react-svg';
import IconCheckMark from '@ergeon/core-components/src/assets/icon-check-mark.svg';
import IconPhoneGreen from '@ergeon/core-components/src/assets/icon-phone-green.svg';

import TellUsForm from '../../containers/TellUsForm';
import { showUpcomingFeatures } from '../../utils/utils';

import { City } from './types';
import { makePhoneLink } from './utils';
import BannerRating from './components/BannerRating';

import './MainBanner.scss';

type MainBannerProps = {
  title: City['header']['title'],
  bullets: City['header']['bullets'],
  phone: City['phone'],
  state: City['state'],
}

const MainBanner = (props: MainBannerProps) => {
  const { title, bullets, phone, state } = props;
  return (
    <section className={`MainBanner is-State${state}`} data-testid='main-banner'>
      <div className="wrapper-1180 flex-wrapper">
        <section className="flex-spacer MainBanner-content">
          <h1 className="h2 MainBanner-title">{title}</h1>
          <ul className="unordered MainBanner-list">
            {bullets.map(bullet => (
              <li className="MainBanner-listItem" key={bullet}>
                <ReactSVG className="icon" src={IconCheckMark} />
                {bullet}
              </li>
            ))}
          </ul>
          <a className="MainBanner-callLink" data-track-call="true" href={makePhoneLink(phone)}>
            <ReactSVG className="icon" src={IconPhoneGreen} />
            <span>Call us now {phone}</span>
          </a>
          {showUpcomingFeatures('ENG-16970') &&
            <BannerRating />
          }
        </section>
        <section className="desktop-length flex-spacer MainBanner-form">
          <TellUsForm />
        </section>
      </div>
    </section>
  )
};

export default MainBanner;
