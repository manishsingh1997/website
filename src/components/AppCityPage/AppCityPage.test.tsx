import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';

import cities from '../../data/cities-full-data.json';
import AppCityPage from './AppCityPage';
import {City} from './types';
import {getAsset} from './utils';
import CityBanner from './CityBanner';
import MainBanner from './MainBanner';

jest.mock('../HomePage/QASection', () => () => <>FAQ</>);
jest.mock('../NotFoundPage', () => () => <>Not found</>);
jest.mock('../../containers/TellUsForm', () => () => <>Tell Us Form</>);

const CITY_BANNER_PHONE = '(543) 094 5738';

describe('AppCityPage', () => {
  it('should render', () => {
    const {container, getAllByTestId} = render(
      <AppCityPage city={cities[0] as unknown as City} />
    );

    const images = getAllByTestId('test-image');
    const image = images[0] as HTMLImageElement;
    expect(container).toMatchSnapshot();
    expect(images).toHaveLength(10);
    expect(image.src).toBe(getAsset(cities[0].packages.data.wood[0].img, 'jpeg'));
  });

  it('Should render city banner', () => {
    const city = 'Coruscant';
    render(<CityBanner city={city} phone={CITY_BANNER_PHONE}/>);

    const cityName = screen.getByText(new RegExp(city, 'ig'));
    const cityBanner = screen.getByTestId(/city-banner/);
    expect(cityName).toBeInTheDocument();
    expect(cityBanner).toMatchSnapshot();
  })

  it('Should render Main city banner', () => {
    render(<MainBanner bullets={['one', 'two']} phone='+1234' state='CA' title='California' />);

    const cityBanner = screen.getByTestId(/main-banner/);
    expect(cityBanner).toMatchSnapshot();
  })
});
