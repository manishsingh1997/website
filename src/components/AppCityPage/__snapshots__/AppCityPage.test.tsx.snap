// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCityPage should render 1`] = `
<div>
  Not found
</div>
`;
